src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -1636,26 +1626,26 @@ abstract class ManagedStrategy(
                 toObject = atomicMethodDescriptor.getSetValue(receiver, params)
             )
         }
-        // check for livelock and create the method call trace point
-        if (collectTrace) {
-            traceCollector?.checkActiveLockDetected()
-            addBeforeMethodCallTracePoint(threadId, receiver, codeLocation, methodId, className, methodName, params,
-                atomicMethodDescriptor, MethodCallTracePoint.CallType.NORMAL
-            )
-        }
+
         // in case of an atomic method, we create a switch point before the method call;
         // note that in case we resume atomic method there is no need to create the switch point,
         // since there is already a switch point between the suspension point and resumption
         if (methodSection == AnalysisSectionType.ATOMIC &&
             // do not create a trace point on resumption
             !(isTestThread(threadId) && isResumptionMethodCall(threadId, className, methodName, params, atomicMethodDescriptor))
         ) {
-            // re-use last call trace point
-            val methodCallTracePoint = callStackTrace[threadId]!!.lastOrNull()?.tracePoint
             newSwitchPoint(threadId, codeLocation, beforeMethodCallSwitch = true)
-            traceCollector?.addTracePointInternal(methodCallTracePoint)
             loopDetector.passParameters(params)
         }
```

Comment:
Could you please rebase on `develop`?
When refactoring loop detector in this PR:
https://github.com/JetBrains/lincheck/pull/656

I made a small change here, to clearly separate cases of atomic/regular method calls.